crates/sequencer/src/serde/state.rs

```diff
@@ -0,0 +1,156 @@
+use std::{fs, path::PathBuf};
+
+use blockifier::state::cached_state::ContractStorageKey;
+use rustc_hash::FxHashMap;
+use serde::{Deserialize, Serialize};
+use starknet_api::{
+    core::{ClassHash, CompiledClassHash, ContractAddress, Nonce},
+    hash::StarkFelt,
+};
+use thiserror::Error;
+
+use crate::state::State;
+
+use super::contract::SerializableContractClass;
+
+/// TODO: wrap the underlying errors in them
+#[derive(Error, Debug)]
+pub enum SerializationError {
+    #[error("{reason:?}")]
+    IoError {
+        reason: String,
+        context: std::io::Error,
+    },
+    #[error("{reason:?}")]
+    SerdeJsonError {
+        reason: String,
+        context: serde_json::Error,
+    },
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, Default)]
+pub struct SerializableState {
+    pub classes: FxHashMap<ClassHash, SerializableClassRecord>,
+    pub compiled_classes_hash: FxHashMap<ClassHash, CompiledClassHash>,
+    pub contracts: FxHashMap<ContractAddress, ClassHash>,
+    #[serde(with = "serialize_contract_storage")]
+    pub storage: FxHashMap<ContractStorageKey, StarkFelt>,
+    pub nonces: FxHashMap<ContractAddress, Nonce>,
+}
+
+impl SerializableState {
+    pub fn dump_state(&self) -> Result<Vec<u8>, SerializationError> {
```

Comment:
I'm wondering: should we rename this? This converts the `SerializableState` into a Vec<u8>, so naming it `dump_state` sounds too specific.

Reply:
how about `serialize_state`?